openhands/utils/import_utils.py

```diff
@@ -21,4 +21,4 @@ def get_impl(cls: type[T], impl_name: str | None) -> type[T]:
         return cls
     impl_class = import_from(impl_name)
     assert cls == impl_class or issubclass(impl_class, cls)
-    return impl_class
+    return cast(type[T], impl_class)
```

Comment:
Hmm do we have to do this? I think this might be implied via the assert statement, and sometimes the `cls` passed is abstract so wonder if doing this cast will affect the existing system